### PR TITLE
Feature/bringup rework

### PR DESCRIPTION
**Pull request by b1n-ch1kn**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimisation

## [required] Description

Refactors how autonomous ROS processes are "brought up" and started.

Highlights:
- Coupled with ACM deployment and uses new AV states - creates new ROS msgs for these.
- Replaces `vehicle supervisor` package and state machine as this state machine has now been moved to the ACM.
- Replaces `mission controller` package as missions are now started through the new supervisor (based on new state messages).
- Moved `QUTMS Embedded Common` submodule from `hardware` group to `common` group and wrapped a ROS package around it for package lookup. 
- Made a `perception bringup` package for all perception related packages - velodyne, ground plane removal, cloud to laserscan, lidar pipeline launches and configs moved there.
- `nav_bringup` package is now more aligned to Nav2 example packages (such as `honeybee_nav2`) with arg definitions.
- `slam_toolbox` launch setup for bringup with lifecycle nodes - allows for control via handler or supervisor nodes. Requires a specific commit between `humble` and `jazzy` to work on Ubuntu 22.04.
- Removed `machine` packages.
- Removed `sensor` package.
- Removed `vision pipeline` package.
- Moved `vehicle urdf` package from `common` group to `operations` group.

## [required] Documentation

Coming soon... ACM and supervisor bringup docs are WIP. 
Deprecation of older packages brings us in line with current stack documentation. 

### [optional] Usability concerns or breaking changes?

Hardware wise this is a breaking change, no longer can older code work with new vehicle hardware in the form of the ACM. Reverting back to the old EBS control board would be required for older code to work now. Specifically, the state machine and mission bringup.

### CODE REVIEW

File: .gitmodules
Comment (leungw16) [inline .gitmodules:1]: Update branch?

**Review by leungw16 — CHANGES_REQUESTED** [review]

File: .gitmodules
Comment (b1n-ch1kn) [reply .gitmodules:1]: 👀 what's this?

Comment (leungw16) [reply .gitmodules:1]: Line 7. Change to current branch name?

Comment (b1n-ch1kn) [reply .gitmodules:1]: good spot
although im not sure if thats ever actually considered? cause the repo's commit gets saved here: https://github.com/QUT-Motorsport/QUTMS_Driverless/pull/329/files#diff-b3f0f4e4c8ef025fd68f6317c74a213cac0c414c8eb9bc4d01f3d7bdff6d4add

**Review by leungw16 — APPROVED** [review]
LGTM

**Pull request merged**